client/src/app/shared/loading-bar/loading-bar.component.ts

```diff
@@ -0,0 +1,20 @@
+import { Component, OnInit, ChangeDetectionStrategy, ChangeDetectorRef, AfterViewInit } from '@angular/core';
+import { LoadingBarService } from './loading-bar.service';
+import { delay } from 'rxjs/operators';
+
+@Component({
+  selector: 'ml-loading-bar',
+  templateUrl: 'loading-bar.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class LoadingBarComponent implements OnInit, AfterViewInit {
+  constructor(private loadingbarservice: LoadingBarService, private _changeDetectorRef: ChangeDetectorRef) {}
+  public _isVisible = false;
+  ngOnInit() {}
+  ngAfterViewInit(): void {
+    this.loadingbarservice.isVisible.subscribe(isVisible => {
```

Comment:
Having in mind that we might want to support "progress" as well, I think it'd be good to introduce an event type enum similar to slim loading bar's: https://github.com/akserg/ng2-slim-loading-bar/blob/master/src/slim-loading-bar.component.ts#L43-L53

With that, we would probably also have rather sth. like `loadingbarservice.events.subscribe(...)`

Quick question @Jefiozie any particular reason this is done in `afterViewInit()` right now?